### DISCUSSION TITLE
File Icons

### DISCUSSION THREAD

**Discussion by yoav-lavi** [community/community#4995] [Other Feature Feedback, Questions, & Ideas]
Hey,
I'd like to suggest adding icons by the file type in the repository file view.
There are [extensions](https://chrome.google.com/webstore/detail/file-icons-for-github-and/ficfmibkjjnpogdcfhfokmihanoldbfe) that already do this but I think it would be beneficial to have this as built in functionality. It's helpful when looking for files of a specific type and as an overview, and has a visual benefit.
Thanks for your consideration!

**Comment by IsaacLf** [comment]
Yes, I came here since I saw Gitlab had icons for the files in their repositories, they're using: [VSCode Material Icon Theme](https://github.com/PKief/vscode-material-icon-theme/), which would very nice to see in Github's repositories 😁

**Comment by Frulfump** [comment]
This would be a great addition. Maybe this should be opened(or moved) in the `Repositories` https://github.com/orgs/community/discussions/categories/repositories category instead of `General` so it becomes clear product feedback.

Hmm lots of dupes in there
https://github.com/orgs/community/discussions/139026
https://github.com/orgs/community/discussions/84959
https://github.com/orgs/community/discussions/43870
https://github.com/orgs/community/discussions/44927
https://github.com/orgs/community/discussions/42347

**Comment by manukumar7** [comment]
I'd like to suggest adding file type icons next to each file in the repository file view. While there are browser extensions that provide this feature, I believe having it built-in would offer several benefits:

✅ Quick visual identification of file types (e.g., JS, MD, JSON, etc.)

🔍 Easier to scan for specific file types at a glance

🎨 Adds a subtle but helpful visual improvement to the UI

It would enhance both usability and the developer experience, especially in large projects with many files.
